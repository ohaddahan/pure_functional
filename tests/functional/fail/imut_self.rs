use functional_macro::pure_functional;

struct Test {
    i1: u32,
}

impl Test {
    #[pure_functional]
    pub fn test(&self) -> u32 {
        self.i1 + 1
    }
}

fn main() {
    let t = Test { i1: 1 };
    assert_eq!(t.test(), 2);
}
