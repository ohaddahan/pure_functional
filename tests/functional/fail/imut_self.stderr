error: pure_functional: function with self is not supported
 --> tests/functional/fail/imut_self.rs:8:5
  |
8 |     #[pure_functional]
  |     ^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `pure_functional` (in Nightly builds, run with -Z macro-backtrace for more info)
